Formula/cling.rb

```diff
@@ -0,0 +1,38 @@
+class Cling < Formula
+  desc "The cling C++ interpreter"
+  homepage "https://root.cern.ch/cling"
+  url "http://root.cern.ch/git/cling.git",
+      :revision => "bfe05be443c3f16b2eff4094384f1bd22480f3bc"
+  version "0.4"
+
+  depends_on "cmake" => :build
+
+  resource "clang" do
+    url "http://root.cern.ch/git/clang.git", :branch => "cling-patches"
+  end
+
+  resource "llvm" do
+    url "http://root.cern.ch/git/llvm.git", :branch => "cling-patches"
+  end
+
+  def install
+    (buildpath/"src").install resource("llvm")
+    (buildpath/"src/tools/cling").install buildpath.children - [buildpath/"src"]
+    (buildpath/"src/tools/clang").install resource("clang")
+    cd buildpath/"src" do
+      mkdir "build" do
+        cd buildpath/"src/build" do
+          system "cmake", *std_cmake_args, "../../src"
+          system "cmake", "--build", "."
```

Comment:
we usually just use `make install` instead of `cmake --build .` and `cmake --build . --target install`